library/src/main/java/com/novoda/downloadmanager/lib/RestartTimeCreator.java

```diff
@@ -0,0 +1,28 @@
+package com.novoda.downloadmanager.lib;
+
+class RestartTimeCreator {
+
+    private final RandomNumberGenerator randomNumberGenerator;
+
+    public RestartTimeCreator(RandomNumberGenerator randomNumberGenerator) {
+        this.randomNumberGenerator = randomNumberGenerator;
+    }
+
+    /**
+     * Returns the time when a download should be restarted.
+     */
+    public long restartTime(FileDownloadInfo fileDownloadInfo, long now) {
```

Comment:
relying on an something external to get the current time doesn't seem very reliable, what about passing in a clock?
https://github.com/novoda/all-4/blob/master/core/src/main/java/com/novoda/all4/download/expiry/ExpiryCalculator.java#L34-L38